docs/usage/shoot_versions.md

```diff
@@ -0,0 +1,108 @@
+# Kubernetes and Machine image versions in Gardener
+
+## Motivation
+
+On the one hand-side Gardener is responsible for managing the Kubernetes and Machine image versions of its Shoot clusters.
+On the other hand-side, Gardener needs to be configured and updated based on the availability and support of the Kubernetes and Machine image version it provides.
+
+For instance the Kubernetes community releases **minor** versions roughly every three months and usually maintains **three minor** versions (the actual and the last two) with bug fixes and security updates. 
+Patch releases are done more frequently.
+
+Gardener needs to be configured accordingly to reflect the "logical state" of a version. 
+It should be possible to define the Kubernetes or Machine image versions that still receive bug fixes and security patches and also vice-versa to define the version that are out-of-maintenance and are potentially vulnerable.
+Moreover, this allows Gardener to "understand" the current state of a version and act upon it (more information in the following sections). 
+
+## Overview
+
+**As a Gardener operator**:
+- I can classify a version based on it's logical state (`preview`, `supported`, `deprecated` and `expired` see [Version Classification](#version-classifications)).
+- I can define which Machine image and Kubernetes versions are eligible for the auto update of Clusters during the maintenance time.
+- I can disallow the creation of clusters having a certain version (think of severe security issues).
+
+**As an end-user/Shoot owner of Gardener**:
+- I can get information which Kubernetes and Machine image versions are supported, in preview, deprecated or expired. 
+- I can determine the time when my Shoot clusters Machine image and Kubernetes version will be forcefully updated to the next patch or minor version (in case the cluster is running a deprecated version with an expiration date).
+- I can get this information via API from the CloudProfile.
+
+## Version Classifications
+
+In their lifetime, versions go through 4 distinct "logical states".
```

Comment:
Can you explain that this is purely the decision of a human operator and could also be done entirely differently, i.e., that what you explain here is just "a best/recommended practice"?